src/redux/books/booksSlice.js

```diff
@@ -1,42 +1,80 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
 
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/6WW9pLyM5KcNpiSeZbhV/books';
 const initialState = {
-  books: [
-    {
-      itemId: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      itemId: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      itemId: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
+  isLoading: true,
 };
 
+export const getBooksItems = createAsyncThunk('books/getBooksItems',
+  async () => {
+    const response = await fetch(url);
+    const data = await response.json();
+    const keys = Object.keys(data);
+    const newBooks = [];
+    let item = {};
+    keys.forEach((e) => {
+      item = {
+        item_id: e,
+        title: data[e][0].title,
+        author: data[e][0].author,
+        category: data[e][0].category,
+      };
+      newBooks.push(item);
+    });
+    return newBooks;
+  });
+
+export const addBook = createAsyncThunk('books/addBook',
+  async (data) => {
+    await fetch(url, {
+      method: 'POST', // or 'PUT'
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    });
+    return data;
+  });
+export const removeBook = createAsyncThunk('books/removeBook',
+  async (id) => {
+    const deleteUrl = `${url}/${id}`;
+    await fetch(deleteUrl, {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json',
+      },
```

Comment:
- [ ] Kindly ensure that all your HTTP requests are made with `axios`. This is a requirement 🙏

Reply:
Hello @codepantha 👋 , thank you for your feedback, however, I was reviewing the requirements and the second says that I can use fetch or axios, I preferred fetch because it worked great for me in the Leaderboard project. I would really appreciate it if you could check it out, here is the link to the [requirements.](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connect_to_api.md)